### ISSUE TITLE
Make the tool for 8.3.1 plugable (currently AIDE)

### ISSUE THREAD

**Issue by memelet** [awailly/cis-ubuntu-ansible#77]
(Thoughts in progress...)

We are currently using OSSEC on all nodes. OSSEC comes with Syscheck, which I think performs the same function as AIDE. Would it be possible to factor out the bit that installs AIDE?

I'm not really sure just how this would work though. At least in our playbooks we run cis first, then ossec after. So file checking won't be installed when cis runs. I could swap the two, but I wanted to run cis as soon as possible.

**Comment by awailly** [comment]
To be clear, you want to perform a level 2 conformity with AIDE disabled ?

Will it suit your need if we set a `with_aide` variable on AIDE related tasks in https://github.com/awailly/cis-ubuntu-ansible/blob/master/tasks/section_08_level2.yml ?

**Comment by systeminsightsbuild** [comment]
Well, I don't want to bypass the conformity, but rather provide for other options than AIDE. I've just started using this role today, and am rebuilding 50 or so AWS nodes. Maybe I will have a more coherent position after some time. For now, I'm just letting AIDE get installed along with Syscheck from OSSEC.

**Comment by awailly** [comment]
Ok. How do you foresee the file integrity module architecture? Did you get some ideas after deploying on your nodes?

**Comment by memelet** [comment]
I'm not close to having a proposal for this. Do you want to close and I will open when I have proposal?

But my thinking would be something along the lines of: A variable to allow for the check of an OSSEC installation with file integrity configured, fail if not. Otherwise install AIDE. 

Installing OSSEC is way to complex and way out of scope to try to install in this role, I would think.

**Comment by awailly** [comment]
No, leave it open. I will come with something when we are ready. I just wanted to fit your needs.

**Comment by memelet** [comment]
Ha! My needs. This project is superb. It was just what I needed and it works like a charm.

**Comment by pchaigno** [comment]
@awailly We could introduce a `use_aide` variable as a first step. What do you think?

**Comment by awailly** [comment]
Yes it was my first idea, renamed to `with_aide`. It is in the queue after 2.0 support.

**Comment by pchaigno** [comment]
> It is in the queue after 2.0 support.

:+1: 

> Yes it was my first idea, renamed to `with_aide`.

We already use `use_apparmor`. We can change but we need to be consistent ;)

**Comment by awailly** [comment]
Absolutely make sense, thanks. I will add the `use_aide` on 8.2.

**Comment by memelet** [comment]
I just started to work on this and I see in section_08_level2.yml

```
  # We have to run the check after AIDE installation as postfix create new matched binaries
  - name: 8.1.12 Collect Use of Privileged Commands (Scored)
```

I'm not sure what the implication of this is. In my playbooks OSSEC (and its file integrity checker) is installed after CIS. I have not seen any problems doing this so far, but maybe I am missing something?

**Comment by awailly** [comment]
The implication is that if we perform the suid binaries check before the AIDE installation, we will miss some sensitive binaries that are installed by AIDE.

Your current workflow with OSSEC will not present any issue if the OSSEC installation does not create files that trigger a new line in the 8.1.12 check. Is the workflow idempotent?

**Comment by systeminsightsbuild** [comment]
Hmm, I run a lot of stuff (different depending on the node's role) after the CIS role. Early on I found some instabilities if run CIS on already built nodes but did not take much time to investigate. But now I wonder if I should be running CIS last. 

All of our playbooks are idempotent. But I'm pretty sure that some files gets installed with permissions different than CIS would desire. Also CIS alters the default permissions, and it running first effects subsequent installs. 

OSSEC performs a continuous CIS audit and I have also run lynis manually -- neither of which have any complaints. But I don't think either of those use the specific ubuntu controls.

**Comment by systeminsightsbuild** [comment]
I used "Resolves" instead of "Closes" in the commit. Does that mean this issue needs to be closed manually?

**Comment by pchaigno** [comment]
[Resolves works](https://help.github.com/articles/closing-issues-via-commit-messages/), it's just that you have to use the following syntax for the issue number: `resolves #77` ;)
